bids/reports/utils.py

```diff
@@ -0,0 +1,573 @@
+"""Generate publication-quality data acquisition methods section from BIDS dataset.
+
+Utilities to generate the MRI data acquisition portion of a
+methods section from a BIDS dataset.
+"""
+from __future__ import print_function
+import re
+import logging
+from os.path import basename
+
+import numpy as np
+import nibabel as nib
+from num2words import num2words
+
+from bids.version import __version__
+
+logging.basicConfig()
+logger = logging.getLogger('pybids.reports.utils')
+
+
+def warnings():
+    """
+    Remind users about things they need to do after generating the report.
+    """
+    return('Remember to double-check everything and to replace <deg> with '
+           'a degree symbol.')
+
+
+def remove_duplicates(seq):
+    """
+    Return unique elements from list while preserving order.
+    From https://stackoverflow.com/a/480227/2589328
+    """
+    seen = set()
+    seen_add = seen.add
+    return [x for x in seq if not (x in seen or seen_add(x))]
+
+
+def num_to_str(num):
+    """
+    Convert an int or float to a nice string.
+    E.g.,
+        21 -> '21'
+        2.500 -> '2.5'
+        3. -> '3'
+    """
+    return '{0:0.02f}'.format(num).rstrip('0').rstrip('.')
+
+
+def list_to_str(lst):
+    """
+    Turn a list into a comma- and/or and-separated string.
+
+    Parameters
+    ----------
+    lst : :obj:`list`
+        A list of strings to join into a single string.
+
+    Returns
+    -------
+    str_ : :obj:`str`
+        A string with commas and/or ands separating th elements from ``lst``.
+
+    """
+    if len(lst) == 1:
+        str_ = lst[0]
+    elif len(lst) == 2:
+        str_ = ' and '.join(lst)
+    elif len(lst) > 2:
+        str_ = ', '.join(lst[:-1])
+        str_ += ', and {0}'.format(lst[-1])
+    else:
+        raise ValueError('List of length 0 provided.')
+    return str_
+
+
+def get_slice_info(slice_times):
```

Comment:
I think the `utils` module is a bit big and contains a bunch of functions that feel to me like they're doing core parsing/extraction stuff. It might make sense to keep `utils` for things like `list_to_str`, and put heavier functions like `get_slice_info` and `get_seqstr` into their own module called something more descriptive like `format` or `parsing` or something. But it's your call.

Reply:
Good idea. I'll move all of the string formatters into `parsing` and keep the actual utilities in `utils`. For now, I'm thinking of keeping `get_slice_info`, `get_seqstr`, and `get_sizestr` in `utils`, mainly because `parsing` still ends up being pretty big even when it only contains the `*_info` functions.